docs: fix permission setup for group-accessible repo

MichaelEischer opened the pull request:
<!--
Thank you very much for contributing code or documentation to restic! Please
fill out the following questions to make it easier for us to review your
changes.
-->

What does this PR change? What problem does it solve?
-----------------------------------------------------

The group always needs execute access for the directories. In addition, files should be always set to read-only for everyone as restic never modifies files.

<!--
Describe the changes and their purpose here, as detailed as needed.
-->

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Noticed while looking at https://github.com/restic/restic/issues/5488 .
<!--
Link issues and relevant forum posts here.

If this PR resolves an issue on GitHub, use "Closes #1234" so that the issue
is closed automatically when this PR is merged.
-->

Checklist
---------

<!--
You do not need to check all the boxes below all at once. Feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box. Enable a checkbox by replacing [ ] with [x].

Please always follow these steps:
- Read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- Enable [maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- Run `gofmt` on the code in all commits.
- Format all commit messages in the same style as [the other commits in the repository](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
-->

- ~~[ ] I have added tests for all code changes.~~
- [x] I have added documentation for relevant changes (in the manual).
- ~~[ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).~~
- [x] I'm done! This pull request is ready for review.

MichaelEischer committed:
docs: fix permission setup for group-accessible repo

The group always needs execute access for the directories. In addition,
files should be always set to read-only for everyone as restic never
modifies files.

MichaelEischer committed:
docs: use absolute permissions for group accessible repositories

brad2014 reviewed (commented):
This sounds much better, but I wanted to mention (as a group access repo creator and user) that I was confused by the original read-only vs read-write conversation.  It may sound odd (and I hope its not inappropriate), but I think the entire section (lines 791-836) might want to be simply something like:

--- suggestion start ---

Since restic version 0.14, both local and SFTP repositories can be made accessible to all the members of a given UNIX group on the repository host.

To permit multiple users to use a repository, first run "restic init" to create it, if it doesn't already exist. Then, some manual intervention is currently required. Run the following commands over the repository files themselves, which give the required permissions (and hints to restic). Thereafter, restic commands can be run against that repository by any member of a given UNIX group.

```
# chgrp -R restic-users /srv/restic-repo
# find /srv/restic-repo -type f -exec chmod 440 '{}' \;
# find /srv/restic-repo -type d -exec chmod 2770 '{}' \;
```

Of course, you'll change `/srv/restic-repo` to the path to your repo and `restic-users` to the name of your UNIX group whose users are allowed access.

(Internally, the group read permission on the `config` file tells restic to create all future ordinary files inside the repo with group-read permission, and the setgid mode bit on directories causes restic to set the group of each newly created file to the group of its parent directory. They thus remain accessible to all group members, regardless of which user created them. Note that restic never creates or deletes directories within an initialized repo, only regular files.)

~~There are a few restic commands that don't create *any* files in the repo, and if you want to further lock down the repo to prevent group members from modifying the repo in any way, just use mode `2750` instead of `2770` above.  This disables group-write access to directories, so files can only be created or deleted by the owner.  Of course, restic subcommands that require write access will fail.~~

For remote repository access via SFTP, note that the user running the restic command on a client machine may be different from the account SFTP is logged in to.  
```
# restic backup -r sftp:restic@repohost:/srv/restic-repo
```
In the example, the local user `root` can read all the files on the client host, and send them for backup to a remote user `restic` to add them to the repository.  In this example, `restic` could be a user in the `restic-users` group on `repohost`.


--- end suggestion ---

(I hope this kind of suggestion, or this forum for it, is not inappropriate. if so, my apologies.)

MichaelEischer committed:
Fully rework docs for group-accessible repositories

Just tell the user what to do instead of explaining too many details.
I've dropped the read-only variant as it actually has no representation
in the local and sftp backends. Instead it relied on both backends
initially creating all directories, which can't actually be guaranteed.

Based on a suggestion by @brad2014 in significant parts.

brad2014 commented:
Thank you.  Looks good :+1:
